src/types/QIX.ts

```diff
@@ -132,14 +131,48 @@ export interface ExtendedDimensionInfo extends EngineAPI.INxDimensionInfo {
   qCardinalities: {
     qHypercubeCardinal: number;
   };
+  columnWidth: ColumnWidth;
+}
+
+export interface ExtendedMeasureInfo extends EngineAPI.INxMeasureInfo {
+  columnWidth: ColumnWidth;
 }
 
-export interface HyperCube extends EngineAPI.IHyperCube {
+export interface ExtendedHyperCube extends Omit<EngineAPI.IHyperCube, "qDimensionInfo" | "qMeasureInfo"> {
+  qDimensionInfo: ExtendedDimensionInfo[];
+  qMeasureInfo: ExtendedMeasureInfo[];
   activelySortedColumn: ActivelySortedColumn;
 }
 
+export interface PivotLayout extends EngineAPI.IGenericHyperCubeLayout {
+  qHyperCube: ExtendedHyperCube;
+  nullValueRepresentation?: NullValueRepresentation;
+  title: string;
+  snapshotData?: SnapshotData;
+  components?: Component[];
+}
+
+export interface SnapshotLayout extends EngineAPI.IGenericObjectLayout {
+  nullValueRepresentation?: NullValueRepresentation;
+  title?: string;
+  snapshotData?: SnapshotData;
+}
+
 export type Model = EngineAPI.IGenericObject | EngineAPI.IGenericBookmark | undefined;
 
 export type App = EngineAPI.IApp | undefined;
 
 export default NxDimCellType;
+
+// types for properties
+interface ExtendedInlineDimensionDef extends EngineAPI.INxInlineDimensionDef {
+  columnWidth: ColumnWidth;
+}
+
+interface ExtendedInlineMeasureDef extends EngineAPI.INxInlineMeasureDef {
+  columnWidth: ColumnWidth;
+}
+
+export interface DimensionOrMeasure {
```

Comment:
Maybe rename to indicate that it's properties. Ex: `DimensionOrMeasureDef` or `DimensionOrMeasureProps`

Reply:
good point